### PR TITLE
Add integration test for issue #933

### PR DESCRIPTION
**Pull request by audn**
This PR adds a new integration test for issue https://github.com/ldapjs/node-ldapjs/issues/933.

### CODE REVIEW

File: test-integration/client/issue-933.test.js [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+         t.error(err, 'bind error');
+      }
+   );
+
+   client.search(baseDN.toString(), opts, (err, res) => {
+      t.error(err, 'search error');
+      res.on('searchEntry', (entry) => {
+         t.match(entry.pojo, {
+            type: 'SearchResultEntry',
+            objectName: 'cn=jdoe,ou=Norge Gjc3\b8vik,dc=planetexpress,dc=com',
```
Comment (jsumners) [inline test-integration/client/issue-933.test.js:38]: This is incorrect. The first byte of the multi-byte character is not escaped, and the escape characters should also be escaped so that they are validated. It should be:
```suggestion
            objectName: 'cn=jdoe,ou=Norge Gj\\c3\\b8vik,dc=planetexpress,dc=com',
```

File: test-integration/client/issue-933.test.js [hunk 2]
```diff
@@ -0,0 +1,54 @@
+'use strict';
+
+const tap = require('tap');
+const ldapjs = require('../../lib');
+const parseDN = ldapjs.parseDN;
+
+const SCHEME = process.env.SCHEME || 'ldap';
+const HOST = process.env.HOST || '127.0.0.1';
+const PORT = process.env.PORT || 1389;
```
Comment (jsumners) [inline test-integration/client/issue-933.test.js:9]:
```suggestion
const PORT = process.env.PORT || 389
```

File: test-integration/client/issue-933.test.js [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+tap.test('can search OUs with Norwegian characters', (t) => {
+   client.bind(
+      'cn=admin,dc=planetexpress,dc=com',
+      'GoodNewsEveryone',
+      (err) => {
+         t.error(err, 'bind error');
+      }
+   );
+
+   client.search(baseDN.toString(), opts, (err, res) => {
```
Comment (jsumners) [inline test-integration/client/issue-933.test.js:33]: The test image does not contain this entry, so you're not going to get a `searchEntry` message. You'll have to add the entry before searching for it. See https://github.com/ldapjs/node-ldapjs/blob/ac588a0fadd156de27622e9dd09101e1bead5d25/test-integration/client/issue-883.test.js

**Review by jsumners — CHANGES_REQUESTED** [review]
Please correct the linting issues.

File: test-integration/client/issue-933.test.js (hunk 3 above)
Comment (audn) [reply test-integration/client/issue-933.test.js]: So this PR is waste? https://github.com/ldapjs/docker-test-openldap/pull/5

I just copied the same steps as you did for https://github.com/ldapjs/node-ldapjs/commit/8c9bfe81f76804e555cf2f0e2cd7b2a8524932fc

Comment (jsumners) [reply test-integration/client/issue-933.test.js]: > So this PR is waste? [ldapjs/docker-test-openldap#5](https://github.com/ldapjs/docker-test-openldap/pull/5)

I had not seen that PR yet. A note in the opening message of this PR that it requires an adjustment to the image would have been helpful.

There a couple things about this approach:

1. That PR would need to be merged and a new image built and uploaded
2. The CI configuration here would need to be updated (https://github.com/ldapjs/node-ldapjs/blob/ac588a0fadd156de27622e9dd09101e1bead5d25/.github/workflows/integration.yml#L23)

> I just copied the same steps as you did for [8c9bfe8](https://github.com/ldapjs/node-ldapjs/commit/8c9bfe81f76804e555cf2f0e2cd7b2a8524932fc)

That test follows the steps outlined above. Adding the entry to the image before writing the test here eliminates any potential issues with adding the entry via ldapjs and isolates the test to searching. It is definitely the correct approach, be we have a problem -- https://github.com/ldapjs/docker-test-openldap/issues/4.

Basically, the `master` branch in the image repo is broken because the upstream OpenLDAP implementation is broken. I have not had time to devise a solution. I will try to do so soon.

Comment (jsumners) [reply test-integration/client/issue-933.test.js]: I decided to take the easy way out and implemented a fix for the described problem in https://github.com/ldapjs/node-ldapjs/pull/935. Please address the feedback in https://github.com/ldapjs/docker-test-openldap/pull/5 and we can move this forward.

**Comment by jsumners** [comment]
I believe we can close this, and the Docker image issue, with release https://github.com/ldapjs/node-ldapjs/releases/tag/v3.0.5. Please try that release and report back.

**Comment by audn** [comment]
Unfortunately, the changes in 3.0.5 did not fix this issue

**Comment by audn** [comment]
I might have got it working, will update soon.

**Comment by jsumners** [comment]
Does 3.0.5 solve this issue?

**Comment by jsumners** [comment]
Given the lack of response, and the fact that the fix included in v3.0.5 includes a test that specifically tests a character included in this issue (https://github.com/ldapjs/filter/commit/e12ec04a3dac543803a57df5ca7cce269355604d#diff-3d1de0fa47b73b302bc2e63ebf2797fe6d17bf7b28a9d369ced5c1d604c011bb), I am closing this (and the other associated issued). 

If you determine that there is still a problem, please reopen.
